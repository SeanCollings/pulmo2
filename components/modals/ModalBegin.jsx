import React, { useContext } from 'react';
import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

import { SELECTED_LEVEL, ProfileContext } from '../../context/profile-context';
import CustomModal from '.';
import { useTheme } from '../../hooks/useTheme';

const ModalBegin = ({ cancelModel, confirmModal, navigation }) => {
  const theme = useTheme();
  const { profileContext } = useContext(ProfileContext);

  const selectedLevel =
    (profileContext && profileContext[SELECTED_LEVEL]) ||
    TOTAL_DIFFICULTY_LEVELS;

  const contents = [
    'Inhale and exhale as slowly and as deeply as you can.',
    'Repeat through all breath cycles.',
    'Rest for the prescribed time.',
    'Repeat for each round.',
    `Current difficulty level: ${selectedLevel}`,
  ];

  const BackButton = () => (
    <TouchableOpacity onPress={cancelModel} style={styles.backButton}>
      <MaterialCommunityIcons
        name="arrow-left"
        color={theme.BORDER}
        size={26}
      />
    </TouchableOpacity>
  );

  const updateLevelHandler = () => {
    navigation.navigate('Profile');
    setTimeout(() => {
      cancelModel();
    }, 0);
  };

  return (
    <CustomModal
      cancelModel={updateLevelHandler}
      confirmModal={confirmModal}
      header={'Ready to begin?'}
      confirmTitle="Begin"
      cancelTitle="Update level"
      headingColour={theme.SECONDARY}
      TopLeft={BackButton}
      buttonStyle={{ height: 60 }}
      animationType="fade"
    >
      <View style={styles.container}>
        {contents.map((content, i) => (
          <View key={i} style={styles.contentContainer}>
            <Text
              style={{
                ...styles.contentText,
                color: theme.TEXT,
                fontFamily:
                  i === contents.length - 1 ? 'tit-regular' : 'tit-light',
                opacity: theme.DARK ? 0.87 : 1,
              }}
            >
              {content}
            </Text>
          </View>
        ))}
      </View>
    </CustomModal>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  contentContainer: {
    paddingTop: 10,
  },
  contentText: {
    textAlign: 'center',
    fontSize: 16,
  },
  backButton: { position: 'absolute', left: 0, paddingTop: 3 },
});

export default ModalBegin;
