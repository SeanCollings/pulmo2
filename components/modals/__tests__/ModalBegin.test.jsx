import React from 'react';
import { act, create } from 'react-test-renderer';
import { cleanup, fireEvent } from 'react-native-testing-library';
import ModalBegin from '../ModalBegin';
import ProfileContextProvider from '../../../context/profile-context';

jest.useFakeTimers();

describe('ModalBegin - unit test', () => {
  let component;

  afterEach(cleanup);

  test('should render default', async () => {
    const cancelModal = () => {};
    const confirmModal = () => {};
    const navigation = () => {};

    await act(async () => {
      component = create(
        <ProfileContextProvider>
          <ModalBegin
            cancelModal={cancelModal}
            confirmModal={confirmModal}
            navigation={navigation}
            animationType="none"
          />
        </ProfileContextProvider>
      );
    });

    expect(component).toMatchSnapshot();
  });

  test('should call updateLevelHandler', async () => {
    const cancelModal = jest.fn();
    const confirmModal = () => {};
    const navigation = { navigate: () => {} };

    await act(async () => {
      component = create(
        <ProfileContextProvider>
          <ModalBegin
            cancelModal={cancelModal}
            confirmModal={confirmModal}
            navigation={navigation}
            animationType="none"
          />
        </ProfileContextProvider>
      );
    });

    const cancelModalButton = component.root.find(
      ({ props }) => props.testID === 'cancelModal'
    );

    fireEvent.press(cancelModalButton);
    expect(cancelModal).toHaveBeenCalled();
  });
});
