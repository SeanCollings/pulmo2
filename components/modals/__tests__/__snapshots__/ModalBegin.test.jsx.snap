// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBegin - unit test should render default 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.5)",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 12,
            "elevation": 5,
            "padding": 20,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 2.84,
            "width": "80%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
              "width": "105%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "left": 0,
                "opacity": 1,
                "position": "absolute",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 33,
                  },
                  undefined,
                  Object {
                    "fontFamily": "material-community",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ÔÅç
            </Text>
          </View>
          <Text
            style={
              Object {
                "color": "#002f56",
                "fontFamily": "tit-regular",
                "fontSize": 20,
                "textAlign": "center",
                "width": "80%",
              }
            }
          >
            Ready to begin?
          </Text>
        </View>
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "paddingTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "paddingTop": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#3e4b57",
                    "fontFamily": "tit-light",
                    "fontSize": 17,
                    "opacity": 1,
                    "textAlign": "center",
                  }
                }
              >
                Inhale and exhale as slowly and as deeply as you can.
              </Text>
            </View>
            <View
              style={
                Object {
                  "paddingTop": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#3e4b57",
                    "fontFamily": "tit-light",
                    "fontSize": 17,
                    "opacity": 1,
                    "textAlign": "center",
                  }
                }
              >
                Repeat through all breath cycles.
              </Text>
            </View>
            <View
              style={
                Object {
                  "paddingTop": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#3e4b57",
                    "fontFamily": "tit-light",
                    "fontSize": 17,
                    "opacity": 1,
                    "textAlign": "center",
                  }
                }
              >
                Rest for the prescribed time.
              </Text>
            </View>
            <View
              style={
                Object {
                  "paddingTop": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#3e4b57",
                    "fontFamily": "tit-light",
                    "fontSize": 17,
                    "opacity": 1,
                    "textAlign": "center",
                  }
                }
              >
                Repeat for each round.
              </Text>
            </View>
            <View
              style={
                Object {
                  "paddingTop": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#3e4b57",
                    "fontFamily": "tit-regular",
                    "fontSize": 17,
                    "opacity": 1,
                    "textAlign": "center",
                  }
                }
              >
                Current difficulty level: 5
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
              "marginTop": 20,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "width": "40%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#CCC",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "paddingHorizontal": 14,
                    "paddingVertical": 8,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.25,
                    "shadowRadius": 2.84,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#3e4b57",
                      "fontFamily": "tit-regular",
                      "fontSize": 14,
                      "opacity": 1,
                      "textAlign": "center",
                      "textTransform": "uppercase",
                    }
                  }
                >
                  Update level
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "width": "40%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#CCC",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "paddingHorizontal": 14,
                    "paddingVertical": 8,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.25,
                    "shadowRadius": 2.84,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#3e4b57",
                      "fontFamily": "tit-regular",
                      "fontSize": 14,
                      "opacity": 1,
                      "textAlign": "center",
                      "textTransform": "uppercase",
                    }
                  }
                >
                  Begin
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
